wgpu/src/lib.rs

```diff
@@ -1809,34 +1833,40 @@ impl Instance {
     /// # Safety
     ///
     /// Refer to the creation of wgpu-core Instance.
-    #[cfg(not(webgpu))]
+    #[cfg(wgpu_core)]
     pub unsafe fn from_core(core_instance: wgc::instance::Instance) -> Self {
         Self {
             context: Arc::new(unsafe {
-                crate::backend::Context::from_core_instance(core_instance)
+                crate::backend::ContextWgpuCore::from_core_instance(core_instance)
             }),
         }
     }
 
     /// Retrieves all available [`Adapter`]s that match the given [`Backends`].
     ///
+    /// Always returns an empty iterator if the instance decided upon creation to
```

Comment:
```suggestion
    /// Always returns an empty vector if the instance decided upon creation to
```

would be nice to still return an iterator, but for that we'd need `itertools::Either`